lib/buffer.js

```diff
@@ -941,6 +941,14 @@ function bidirectionalIndexOf(buffer, val, byteOffset, encoding, dir) {
   }
   dir = !!dir;  // Cast to bool.
 
+  const isUtf8 = encoding === 'utf8' || encoding === 'utf-8' || encoding === undefined;
+  if (isUtf8 && typeof val === 'number') {
+    const method = dir ? 'indexOf' : 'lastIndexOf';
+    let value = (val > 128 || val < -128) ? new Uint8Array([val])[0] : val;
```

Comment:
I think that `new Uint8Array([val])[0]` destroys perf. As we have no improvement after running the buffers benchmark (`buffer-indexof-number.js` -89% 😨)

But I didn't really understand how I could cast value into uint8 properly.

Reply:
I can probably do this with a bitwise operator (probably a mask)
<img width="373" alt="Screenshot 2022-11-25 at 21 23 18" src="https://user-images.githubusercontent.com/22824417/204052233-d8d49353-beae-4c76-a3dd-c3d6e55f06b7.png">
<img width="366" alt="Screenshot 2022-11-25 at 21 23 23" src="https://user-images.githubusercontent.com/22824417/204052235-481a7eb4-f50c-4fff-bc49-e54d760a595a.png">

Reply:
`-140 & 0b11111111`  -> 116 🎉